examples/solvers/bregman_tv_tomography.py

```diff
@@ -0,0 +1,134 @@
+"""Example of Bregman-TV in tomography.
+
+The example solves an inverse problem in tomography using Bregman iterations
+for a total variation (TV) regularized problem. To this end, let ``A`` denote
+the forward operator, ``g`` denote the given data, ``D(Ax, g)`` denote data
+discrepancy functional, and ``alpha * R(x)`` denote the regularizing
+functional. In this particular example, the variational problem considered is
+the TV-problem
+
+    min_x  1/2 ||A(x) - g||_2^2 + lam || |grad(x)| ||_1.
+
+For a functional ``F`` the Bregman functional is defined as
+
+    D_F^p(x, z) = F(x) - F(z) - <p, x-z>,
+
+where ``p`` is a subgradient at the point ``x``, i.e., ``p in dF(x)``.
+
+The Bregman iterations can be written as
+
+    x_{k+1} in argmin D(Ax, g) + lam * D_R^{p_k}(x, x_k)
+    p_{k+1} - p_k in -(1/lam) * A^* dD(Ax_k, g).
+
+Note that the first step involves solving an optimization problem, which is
+(normally) done using in iterative method. This means that the solution method
+contains a set of inner and outer iterations.
+
+The Bregman iterations need to be stopped after a finite number of outer
+iterations. Otherwise the noise from ``g`` will start to deteriorate the
+result. In this example this is illustrated by doing a few outer iterations to
+many, and displaying the result after each outer iteration.
+
+For more details on Bregman iterations, see, e.g., [Bur2016].
+
+Referenses
+----------
+[Bur2016] Burger, M. *Bregman distances in inverse problems and partial
+differential equations*. In: Hiriart-Urruty, J B, Korytowski A, Maurer H,
+Szymkat M. Advances in Mathematical Modeling, Optimization and Optimal Control.
+Springer, 2016, pp 3--33.
+https://doi.org/10.1007/978-3-319-30785-5_2
+https://arxiv.org/abs/1505.05191
+"""
+
+import odl
+
+# Reconstruction space
+reco_space = odl.uniform_discr(min_pt=[-20, -20], max_pt=[20, 20],
+                               shape=[128, 128], dtype='float32')
+
+# Make a parallel beam geometry with flat detector
+geometry = odl.tomo.parallel_beam_geometry(reco_space)
+
+# Create the ray transform
+ray_trafo = odl.tomo.RayTransform(reco_space, geometry)
+
+# Create phantom
+discr_phantom = odl.phantom.shepp_logan(reco_space, modified=True)
```

Comment:
Can you try to make larger blocks of code with comments on top? This feels like lots of comments that state the obvious. For instance, the phantom could easily be part of the "create data" block, and the reco space, geometry and ray trafo could be one block as well.